src/ConnectedMode/Suppressions/SuppressionIssueStoreUpdater.cs

```diff
@@ -30,16 +37,121 @@ internal interface ISuppressionIssueStoreUpdater
+        public SuppressionIssueStoreUpdater(ISonarQubeService server,
+            IServerQueryInfoProvider serverQueryInfoProvider,
+            IServerIssueStoreWriter storeWriter,
+            ILogger logger)
+            : this(server, serverQueryInfoProvider, storeWriter, logger, ThreadHandling.Instance)
+        {
+        }
+
+        internal /* for testing */ SuppressionIssueStoreUpdater(ISonarQubeService server,
+            IServerQueryInfoProvider serverQueryInfoProvider,
+            IServerIssueStoreWriter storeWriter,
+            ILogger logger,
+            IThreadHandling threadHandling)
+        {
+            this.server = server;
+            this.serverQueryInfoProvider = serverQueryInfoProvider;
+
+            this.storeWriter = storeWriter;
+            this.logger = logger;
+            this.threadHandling = threadHandling;
+        }
+
+        #region ISuppressionIssueStoreUpdater
+
+        public async Task UpdateAllServerSuppressionsAsync()
+        {
+            await threadHandling.SwitchToBackgroundThread();
+
+            try
+            {
+                logger.WriteLine(Resources.Suppressions_Fetch_AllIssues);
+
+                CancelCurrentOperation();
+                var localCopyOfCts = updateAllCancellationTokenSource = new CancellationTokenSource();
+                
+                (string projectKey, string serverBranch) queryInfo = await serverQueryInfoProvider.GetProjectKeyAndBranchAsync(localCopyOfCts.Token);
+
+                localCopyOfCts.Token.ThrowIfCancellationRequested();
+
+                if (queryInfo.projectKey == null || queryInfo.serverBranch == null)
+                {
+                    return;
+                }
+
+                localCopyOfCts.Token.ThrowIfCancellationRequested();
+
+                var allSuppressedIssues = await server.GetSuppressedIssuesAsync(queryInfo.projectKey, queryInfo.serverBranch, localCopyOfCts.Token);
+                storeWriter.AddIssues(allSuppressedIssues, clearAllExistingIssues: true);
+
+                logger.WriteLine(Resources.Suppression_Fetch_AllIssues_Finished);
+            }
+            catch (OperationCanceledException)
+            {
+                logger.WriteLine(Resources.Suppressions_FetchOperationCancelled);
+            }
+            catch(Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+            {
+                logger.LogVerbose(Resources.Suppression_FetchError_Verbose, ex);
+                logger.WriteLine(Resources.Suppressions_FetchError_Short, ex.Message);
+            }
+        }
+
+        public Task UpdateServerSuppressionsAsync(DateTimeOffset fromTimestamp)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Clear()
+        {
+            throw new NotImplementedException();
+        }
+
+        private void CancelCurrentOperation()
+        {
+            // We don't want multiple "fetch all" operations running at once (e.g. if the
```

Comment:
nit pick but I think you missed the word "user" in the sentence -> (e.g. if the opens a solution then clicks "update").